packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -487,11 +540,40 @@ export function makeVirtualObjectManager(
       if (initializing) {
         cache.remember(innerSelf);
       }
-      const representative = Far(tag, actualize(wrappedData));
+      const self = actualize(wrappedData);
+      let representative;
+      let exposedRepresentative;
+      const facetiousness = assessFacetiousness(self);
+      switch (facetiousness) {
+        case 'one': {
+          representative = Far(tag, self);
+          vrm.checkOrAcquireFacetNames(kindID, null);
+          exposedRepresentative = representative;
+          break;
+        }
+        case 'many': {
+          exposedRepresentative = {};
+          representative = [];
+          const facetNames = Object.getOwnPropertyNames(self).sort();
+          vrm.checkOrAcquireFacetNames(kindID, facetNames);
+          for (const facetName of facetNames) {
+            const facet = Far(`${tag} ${facetName}`, self[facetName]);
+            exposedRepresentative[facetName] = facet;
+            representative.push(facet);
+          }
+          harden(exposedRepresentative);
+          break;
+        }
+        case 'not':
+          assert.fail(`invalid self actualization for ${tag}`);
```

Comment:
```suggestion
          assert.fail(X`invalid self actualization for ${q(tag)}`);
```
This is a user provided string, so better to quote it into the error message so that it's harder for it to create a misleading one.